### PR TITLE
Strip Linux binaries

### PR DESCRIPTION
**Pull request by abarth**
We distribute some Linux binaries. It's useful for them to be small.

### CODE REVIEW

File: build/toolchain/gcc_toolchain.gni [hunk 1]
```diff
@@ -185,9 +185,21 @@ template("gcc_toolchain") {
...
 
     tool("link") {
-      outfile = "{{root_out_dir}}/{{target_output_name}}{{output_extension}}"
+      exename = "{{target_output_name}}{{output_extension}}"
+      outfile = "{{root_out_dir}}/$exename"
       rspfile = "$outfile.rsp"
-      command = "$ld {{ldflags}} -o $outfile -Wl,--start-group @$rspfile {{solibs}} -Wl,--end-group $libs_section_prefix {{libs}} $libs_section_postfix"
+      unstripped_outfile = outfile
+
+      if (defined(invoker.strip)) {
```
Comment (jamesr) [inline build/toolchain/gcc_toolchain.gni]: add a section to the top-of-file comment documenting what the "strip" variable does.  in particular, it's useful to document exactly which files end up where when this is set or not set.  you can crib the one from chromium

**Comment by abarth** [comment]
#1237

File: build/toolchain/gcc_toolchain.gni (hunk 1 above)
Comment (abarth) [inline build/toolchain/gcc_toolchain.gni]: Done.

File: build/toolchain/linux/BUILD.gn [hunk 2]
```diff
@@ -80,6 +80,7 @@ gcc_toolchain("clang_x64") {
   nm = "nm"
   ar = "ar"
   ld = cxx
+  strip = "strip"
```
Comment (jamesr) [inline build/toolchain/linux/BUILD.gn]: you can really simplify the android build too if you use this mechanism instead of what's here:

https://github.com/flutter/engine/blob/master/build/toolchain/android/BUILD.gn#L62

but that might be better as a separate patch.

**Comment by jamesr** [comment]
lgtm

**Comment by abarth** [comment]
Yeah, I'll do the android thing separately...
